src/wallet/rpcwallet.cpp

```diff
@@ -2626,17 +2626,16 @@ UniValue listwallets(const JSONRPCRequest& request)
             "Returns a list of currently loaded wallets.\n"
             "For full information on the wallet, use \"getwalletinfo\"\n"
             "\nResult:\n"
-            "[                         (json array of strings)\n"
-            "  \"walletname\"            (string) the wallet name\n"
-            "   ...\n"
-            "]\n"
+            "{\n"
```

Comment:
This is breaking change. Also, this is not so relevant to this PR? I suggest to create a PR with this to separate discussions.

Reply:
Another interface is:
```json
{
  "wallets": [
    { "name": "foo", "loaded": true },
    { "name": "bar", "loaded": false }
  ]
}
```
Which has the advantage of allowing to add new attributes later.

Reply:
Re: breaking change, I believe it was only added in 0.15 anyway so its expected to be more unstable. Can separate into a new PR if desired

Reply:
Also re: interface, that idea is nice but it doesn't work well with the current logic of looking for `*.dat` files imo, `available` is slightly less suggestive that they *are* actually wallets than putting them all in a `wallets` list, you'd end up with `{ "name": "peers.dat", "loaded": false}` if you were using the datadir. If you or anyone has a better idea for finding wallets then that would be ok

Reply:
> If you or anyone has a better idea for finding wallets then that would be ok

You could look for berkeleydb magic bytes to at least filter out files which can't be bdb databases:

https://github.com/file/file/blob/master/magic/Magdir/database

Reply:
What is the goal? Why anyone wants to list available wallets? I guess that whoever wants multiwallet keeps a record of the wallets externally? Anyway, I strongly suggest to move that discussion to a new PR.

Reply:
This change may have been influenced by my comment here: https://github.com/bitcoin/bitcoin/issues/11348#issuecomment-330234723. See the original post in that issue for motivation.

My view is that `listwallets` was announced as experimental in v0.15 (along with all the other multiwallet RPCs) so we have a bit more licence to make breaking changes than normal. I think this change should stay, but like @MeshCollider says, there may be better ways to achieve it.